src/rgw/rgw_kafka.cc

```diff
@@ -206,10 +208,18 @@ connection_ptr_t& create_connection(connection_ptr_t& conn) {
     if (!conn->user.empty()) {
       // use SSL+SASL
       if (rd_kafka_conf_set(conn->temp_conf, "security.protocol", "SASL_SSL", errstr, sizeof(errstr)) != RD_KAFKA_CONF_OK ||
-              rd_kafka_conf_set(conn->temp_conf, "sasl.mechanism", "PLAIN", errstr, sizeof(errstr)) != RD_KAFKA_CONF_OK ||
               rd_kafka_conf_set(conn->temp_conf, "sasl.username", conn->user.c_str(), errstr, sizeof(errstr)) != RD_KAFKA_CONF_OK ||
               rd_kafka_conf_set(conn->temp_conf, "sasl.password", conn->password.c_str(), errstr, sizeof(errstr)) != RD_KAFKA_CONF_OK) goto conf_error;
       ldout(conn->cct, 20) << "Kafka connect: successfully configured SSL+SASL security" << dendl;
+
+      if (conn->mechanism) {
+        if (rd_kafka_conf_set(conn->temp_conf, "sasl.mechanism", conn->mechanism->c_str(), errstr, sizeof(errstr)) != RD_KAFKA_CONF_OK) goto conf_error;
+        ldout(conn->cct, 20) << "Kafka connect: successfully configured SASL mechanism" << dendl;
+      } else {
+        if (rd_kafka_conf_set(conn->temp_conf, "sasl.mechanism", "PLAIN", errstr, sizeof(errstr)) != RD_KAFKA_CONF_OK) goto conf_error;
```

Comment:
On the kafka side, SASL can be used without SSL. But on the rgw side so far, user/password must be provided over HTTPS?

> **from doc:** user/password may only be provided together with use-ssl, if not, the connection to the broker would fail.

Reply:
@huynp1999  you are right. looked at the code, ad we actually do not support "SASL_PLAIN"
any SASL authentication requires two things:
1) connection between S3 client and the RGW should be secure. you can workaround that (with this fix: https://github.com/ceph/ceph/pull/43436) by setting `rgw_allow_notification_secrets_in_cleartext` to `true`
2) the connection between the RGW and the kafka broker should be secure

if we do want to support "SASL_PLAIN" I would recommend that we use the above conf parameter this condition:
https://github.com/ceph/ceph/blob/main/src/rgw/rgw_kafka.cc#L523

you can do this as part of this PR, or we can do this as a separate work item (tracker+PR).